services/src/test/java/io/scalecube/services/ServiceUnitTest.java

```diff
@@ -0,0 +1,134 @@
+package io.scalecube.services;
+
+import static org.junit.Assert.assertTrue;
+
+import io.scalecube.services.ServicesConfig.Builder.ServiceConfig;
+import io.scalecube.services.annotations.AnnotationServiceProcessor;
+import io.scalecube.services.annotations.ServiceProcessor;
+import io.scalecube.services.routing.RandomServiceRouter;
+import io.scalecube.services.routing.Router;
+import io.scalecube.services.routing.RouterFactory;
+import io.scalecube.transport.Address;
+import io.scalecube.transport.Message;
+
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Optional;
+
+public class ServiceUnitTest {
+
+  @Test
+  public void service_headers_unit_test() {
+    Message request = Message.builder().header(ServiceHeaders.METHOD, "m")
+        .header(ServiceHeaders.SERVICE, "s")
+        .header(ServiceHeaders.SERVICE_REQUEST, "req")
+        .header(ServiceHeaders.SERVICE_RESPONSE, "res").build();
+
+    assertTrue(ServiceHeaders.serviceMethod(request).equals("m"));
+    assertTrue(ServiceHeaders.service(request).equals("s"));
+    assertTrue(ServiceHeaders.serviceRequest(request).equals("req"));
+    assertTrue(ServiceHeaders.serviceResponse(request).equals("res"));
+  }
+
+  @Test
+  public void test_local_service_instance() {
+
+    LocalServiceInstance instance = new LocalServiceInstance(new ServiceConfig(new GreetingServiceImpl()),
+        "a", "b", new HashMap<>());
+    assertTrue(instance.toString().equals("LocalServiceInstance [serviceObject=GreetingServiceImpl [], memberId=a]"));
+    assertTrue(instance.tags().isEmpty());
+    assertTrue(instance.memberId().equals("a"));
+    assertTrue(instance.serviceName().equals("b"));
+  }
+
+  @Test
+  public void test_remote_service_instance() {
+
+    Microservices member = Microservices.builder().build();
+    ServiceReference reference = new ServiceReference("a", "b", Address.create("localhost", 4000));
+    ServicesConfig config = ServicesConfig.empty();
+
+    ServiceProcessor processor = new AnnotationServiceProcessor();
+    ServiceRegistry registry = new ServiceRegistryImpl(member.cluster(), config, processor);
+    RemoteServiceInstance instance = new RemoteServiceInstance(registry, reference, new HashMap<>());
+
+    assertTrue(instance.toString().equals("RemoteServiceInstance [address=localhost:4000, memberId=a]"));
+    assertTrue(instance.tags().isEmpty());
+    assertTrue(instance.memberId().equals("a"));
+    assertTrue(instance.address().equals(Address.create("localhost", 4000)));
+    assertTrue(!instance.isReachable());
+    assertTrue(!instance.isLocal());
+    assertTrue(instance.serviceName().equals("b"));
```

Comment:
I think for such cases it is better to use `assertEquals` instead of `assertTrue` - it will give much more helpful error message in case if assert won't pass.